--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -90,7 +90,7 @@
 option(WITH_THREADING "Include threading support?" ON)
 if(WITH_THREADING)
 	add_definitions("-DWITH_THREADING")
-	if(WIN32)
+	if(MSVC)
 		find_package(PThreads4W REQUIRED)
 	endif()
 endif()
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -108,7 +108,7 @@
 endif()
 
 if(WITH_THREADING)
-	if(WIN32)
+	if(MSVC)
 		set (LIBRARIES ${LIBRARIES} PThreads4W::PThreads4W)
 	else()
 		set(THREADS_PREFER_PTHREAD_FLAG ON)
--- a/apps/mosquitto_ctrl/CMakeLists.txt
+++ b/apps/mosquitto_ctrl/CMakeLists.txt
@@ -106,7 +106,7 @@
 	)
 
 	if(WITH_THREADING)
-		if(WIN32)
+		if(MSVC)
 			target_link_libraries(mosquitto_ctrl PRIVATE PThreads4W::PThreads4W)
 		else()
 			set(THREADS_PREFER_PTHREAD_FLAG ON)
